### ISSUE TITLE
CUVID hardware acceleration doesn't work

### ISSUE THREAD

**Issue by RobbedColek** [fsinapsi/ffmatch#3]
Hello,

Firstly, I want to say big thank you for developing this software, it's been a great help with managing my video collection so far.

I am having issue with getting CUVID to work though, hardware acceleration in FFMPEG works fine for me, but I can't get it to work at all with ffmatch - all I get is an error.

Is this possibly an issue with included libav DLLs in the release?

Best regards,
Piotr

**Comment by RobbedColek** [comment]
Just an update - my idea was right. Replacing the libav DLLs fixed the issue!

**Comment by fsinapsi** [comment]
> Just an update - my idea was right. Replacing the libav DLLs fixed the issue!

Please be careful, CUVID support in ffmatch is not reliable (the order of decoded frames may be incorrect).
Unfortunately I can't test it.

**Comment by RobbedColek** [comment]
Yes, I've tried it with few movies and I think it works fine when automatic scene matching is running. However, once you have to do frame-seeking manually, the frames are incorrect (on scene changes you'll get some duplicate frames for example).

I'll try to do some more tests and see if I get same exact results when running automatic matching with and without CUVID - if so, it would be nice if that's possible to fallback to software decoding after automatic matching is done for manual matching, as hardware decoding is way faster when scanning the file.

**Comment by fsinapsi** [comment]
> Yes, I've tried it with few movies and I think it works fine when automatic scene matching is running. However, once you have to do frame-seeking manually, the frames are incorrect (on scene changes you'll get some duplicate frames for example).

Thanks for the feedback.
I added the CUVID option at a user's request, but I don't have a GeForce, so I can't test it.
From the feedbacks I understand that it is not reliable, so please don't use it.

At the moment, CUVID support is simply in lines 432 and 435 of this file:

https://github.com/fsinapsi/TreeP/blob/master/trpavcodec/trpavcodec.c

It simply replaces the avcodec_find_decoder_by_name("h264_cuvid") (or "hevc_cuvid") call to avcodec_find_decoder()

I believe this is not enough, but I don't know the correct way.

I read this thread:

https://stackoverflow.com/questions/52875455/how-to-execute-a-ffmpeg-code-on-a-gpu-without-using-the-command-line

but unfortunately it is limited to what I have already done and is not enough.

Bye
